File: server/model/Category.js
'use strict';

let bookshelf = require('./bookshelf');
let BaseModel = require('./BaseModel');

require('./Product');

let Category = BaseModel.extend({
    tableName: 'category',
    attrs: {
    	id: {
    		field: 'id'
    	},
    	name: {
    		field: 'name'
    	},
    	parentId: {
    		field: 'parent_id'
    	},
    	order: {
    		field: 'order'
    	},
    	remark: {
    		field: 'remark'
    	},
        status: {
            field: 'status'
        },
        createAt: {
            field: 'create_at'
        },
        updateAt: {
            field: 'update_at'
        }
    },
    products: function() {
        return this.belongsToMany('Product');
    }
}, {

});

Category.STATUS_OPEN  = 1;
Category.STATUS_CLOSE = 2;

module.exports = bookshelf.model('Category', Category);

File: server/model/category.go
package model

import (
    "github.com/jinzhu/gorm"
)

// Category 商品分类
type Category struct {
    gorm.Model
    Name     string `json:"name"`
    Order    int    `json:"order"`
    ParentID int    `json:"parentId"`
    Status   int    `json:"status"`     
    Remark   string `json:"remark"`
}

// 开启
const CategoryStatusOpen  = 1

// 关闭
const CategoryStatusClose = 2